addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -63,6 +65,49 @@ const KnowledgeArticleFormController = FormController.extend({
         });
     },
 
+    /**
+     * @override
+     * @returns {Promise<boolean>}
+     */
+    canBeDiscarded: function () {
```

Comment:
It should be `canBeRemoved`. We should also save the changes if the user is an admin...